src/lib.rs

```diff
@@ -686,10 +784,7 @@ pub struct SubscriptionDeadlineIterator {
 impl Iterator for SubscriptionDeadlineIterator {
     type Item = Message;
     fn next(&mut self) -> Option<Self::Item> {
-        match self.r.recv_timeout(self.to) {
-            Ok(m) => Some(m),
-            _ => None,
-        }
+        self.r.recv_timeout(self.to).ok()
```

Comment:
This is a handy built-in method for "downgrading" a Result to an Option

Reply:
Nice to know!